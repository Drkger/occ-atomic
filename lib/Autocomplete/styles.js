import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
