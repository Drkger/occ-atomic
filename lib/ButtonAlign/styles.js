export default {
    sm: {
        marginTop:'19px'
    },
    md: {
        marginTop:'9px'
    },
    lg: {
        marginTop:'3px'
    }
};