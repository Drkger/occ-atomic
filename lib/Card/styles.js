import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import shadows from '../subatomic/shadows';

const styles = {
    card: {
        background:colors.white,
        borderRadius:'5px',
        padding:'30px 20px'
    },
    cardNoPadding: {
        extend: 'card',
        padding:'0'
    }
};

const pushShadows = function() {
    for (let i = 0; i<7; i++) {
        styles[`shadow${i}`] = {
            boxShadow: shadows[`lvl${i}`]
        };
    }
};

pushShadows();

export default styles;