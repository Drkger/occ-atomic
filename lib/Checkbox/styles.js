import colors from '../subatomic/colors';
import spacing from '../subatomic/spacing';
import icons from '../subatomic/icons';

export default {
    cont: {
        paddingTop:spacing.tiny,
        paddingBottom:spacing.tiny,
        boxSizing:'border-box',
        display:'flex',
        alignItems:'center',
        cursor:'pointer',
        outline:'0',
        '&:after': {
            content:'""',
            display:'table',
            clear:'both'
        },
        '&:hover $check:after': {
            background:icons.base(icons.check.icon([colors.grey200]))
        }
    },
    check: {
        width:spacing.base,
        height:spacing.base,
        position:'relative',
        '&:before': {
            content:'""',
            width:spacing.small,
            height:spacing.small,
            borderRadius:4,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            border:`1px solid ${colors.grey200}`,
            background:colors.bgWhite
        },
        '&:after': {
            content:'""',
            width:18,
            height:18,
            marginTop:-1,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            transition:'0.3s all',
            background:'transparent'
        }
    },
    active: {
        '& $check:after': {
            background:[icons.base(icons.check.icon([colors.prim])), '!important']
        }
    },
    disabled: {
        opacity:0.4,
        pointerEvents:'none'
    },
    label: {
        marginLeft:spacing.tiny,
        cursor:'pointer'
    }
};
