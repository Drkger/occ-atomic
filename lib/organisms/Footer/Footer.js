import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';
import $ from 'jquery';

import Container from '../../elements/layout/Container';
import Icon from '../../elements/Icon';

class Footer extends React.Component {
    
    showSub(button) {
        $('#'+button.sub.id).fadeIn('fast').click((e) => {
            e.stopPropagation();
        });
        $('#'+button.sub.id+' a').click(function() {
            $('#'+button.sub.id).fadeOut('fast');
        });
        $('body').click(function() {
            $('#'+button.sub.id).fadeOut('fast');
        });
    }
    
    render() {
        const { classes, buttons } = this.props;
        return (
            <Container fluid={true} className={classes.footer}>
                <ul className={classes.footerList}>
                    {
                        buttons.map(button => (
                            <li key={button.id} className={classes.listElement}>
                                <a className={classes.link} id={button.id} href={button.href} target={button.target} onClick={'sub' in button ? () => {this.showSub(button);} : button.onClick ? button.onClick : undefined}>{button.label}</a>
                                {button.sub && (
                                    <ul className={classes.dropdownMenu} id={button.sub.id} style={button.sub.style}>
                                        {button.sub.items.map(item => (
                                            <li key={item.id} className={classes.dropdownElement}>
                                                <a className={classes.dropdownLink} id={item.id} href={item.href} target={item.target}>{item.label}</a>
                                            </li>
                                        ))}
                                    </ul>
                                )}
                            </li>
                        ))
                    }
                </ul>
            </Container>
        );
    }
}

Footer.propTypes = {
    classes: PropTypes.object,
    buttons: PropTypes.array
};

export default Footer;