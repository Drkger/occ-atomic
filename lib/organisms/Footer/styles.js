import colors from '../../subatomic/colors';
import fonts from '../../subatomic/fonts';
import grid from '../../subatomic/grid';
import icons from '../../subatomic/icons';

export default {
    footer: {
        backgroundColor:colors.black,
        height:'25px',
        margin:'0',
        padding:'0 40px',
        fontSize:'12px',
        color:colors.grey2,
        fontFamily:fonts.title
    },
    footerList: {
        listStyle:'none',
        width:'auto',
        margin:'0',
        padding:'4px 0 0 0'
    },
    listElement: {
        listStyle:'none',
        float:'left',
        margin:'0 50px 0 0',
        padding:'0',
        position:'relative'
    },
    link: {
        color:colors.grey2,
        cursor:'pointer',
        textDecoration:'none',
        '&:hover, &:active, &:focus': {
            color:colors.white,
            textDecoration:'none'
        }
    },
    [`@media (max-width: ${grid.md-1}px)`]: {
        footer: {
            display: 'none !important'
        }
    },
    dropdownMenu: {
        display:'none',
        background:colors.black,
        position:'absolute',
        bottom:'calc(100% + 10px)',
        left:'50%',
        transform:'translateX(-50%)',
        minWidth:'160px',
        padding:'5px 0',
        fontSize:'12px',
        textAlign:'left',
        listStyle:'none',
        borderRadius:'4px',
        boxShadow:'0 6px 12px rgba(0,0,0,.175)',
        zIndex:'2'
    },
    dropdownElement: {
        float:'left',
        listStyle:'none'
    },
    dropdownLink: {
        color:colors.grey2,
        textDecoration:'none',
        padding:'3px 18px',
        lineHeight:'1.42857143',
        display:'block',
        '&:hover': {
            color:colors.white,
            textDecoration:'none'
        }
    }
};