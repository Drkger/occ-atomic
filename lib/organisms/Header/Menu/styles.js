import colors from '../../../subatomic/colors';
import grid from '../../../subatomic/grid';
import icons from '../../../subatomic/icons';

export default {
    menu: {
        marginTop:'5px',
        display:'inline-block',
        cursor:'pointer !important',
    },
    btnContainer: {
        position:'relative',
        display:'inline-block',
        verticalAlign:'middle'
    },
    btnMenu: {
        cursor:'pointer',
        background:'none',
        border:'none',
        padding:'5px 10px 2px 10px',
        marginLeft:'10px',
        color:colors.white,
        transition:'0.3s all',
        outline:'none !important',
        borderRadius:'3px',
        '&:hover': {
            color:colors.blue,
            background:colors.white
        },
        '&:hover $menuIcon': {
            background:icons.base(icons.menu.icon(["#1476fb"]))
        }
    },
    btnMenuOpen: {
        color:colors.blue,
        background:colors.white,
        '& $menuIcon': {
            background:icons.base(icons.menu.icon(["#1476fb"]))
        }
    },
    userName: {
        paddingRight:'5px',
        margin:'0',
        display:'inline-block',
        overflow:'hidden',
        textOverflow:'ellipsis',
        maxWidth:'65px',
        whiteSpace:'nowrap'
    },
    menuIcon: {
        width:icons.menu.width,
        height:icons.menu.height,
        display:icons.menu.display,
        transition:'0.3s all',
        background:icons.base(icons.menu.icon(["#FFFFFF"]))
    },
    dropdownMenu: {
        position:'absolute',
        top:'100%',
        zIndex:'1000',
        display:'none',
        float:'left',
        minWidth:'160px',
        padding:'5px 0',
        fontSize:'14px',
        textAlign:'left',
        listStyle:'none',
        backgroundColor:colors.white,
        backgroundClip:'padding-box',
        border:'1px solid rgba(0,0,0,.15)',
        boxShadow:'0 6px 12px rgba(0,0,0,.175)',
        margin:'13px 0 0 -187px',
        right:'0'
    },
    dropdownMenuOpen: {
        display:'block'
    },
    listAnchor: {
        textDecoration:'none',
        display:'block',
        padding:'0 10px',
        margin:'5px 0 5px 0',
        clear:'both',
        fontWeight:'400',
        lineHeight:'1.42857143',
        whiteSpace:'nowrap',
        color:'#000',
        cursor:'pointer',
        '&:hover, &:focus': {
            textDecoration:'none',
            color:colors.blue
        }
    },
    signOut: {
        color:`${colors.pink} !important`,
        margin:'10px 0 5px 0'
    },
    separator: {
        height: '1px',
        background:'#ebebeb'
    },
    [`@media (max-width: ${grid.sm-1}px)`]: {
        dropdownMenu: {
            width:'200px'
        }
    }
};