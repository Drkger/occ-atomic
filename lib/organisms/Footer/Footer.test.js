import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';
import styles from './styles';
import FooterJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Footer", () => {

    it('matches the snapshot', () => {
        const buttons = [
            {
                id: 'item1',
                href:'http://occ.com.mx',
                label:'Item 1'
            }
        ];
        const wrapper = shallow(
            <Footer buttons={buttons} classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("FooterJSS", () => {

    it('matches the snapshot', () => {
        const buttons = [
            {
                id: 'item1',
                href:'http://occ.com.mx',
                label:'Item 1'
            }
        ];
        const wrapper = shallow(
            <FooterJSS buttons={buttons} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});