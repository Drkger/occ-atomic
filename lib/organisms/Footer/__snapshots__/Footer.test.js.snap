// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    buttons={
      Array [
        Object {
          "href": "#",
          "id": "item1",
          "label": "Item 1",
        },
        Object {
          "href": "#",
          "id": "item2",
          "label": "Item 2",
        },
        Object {
          "id": "item3",
          "label": "Item 3",
          "sub": Object {
            "id": "item3sub",
            "items": Array [
              Object {
                "href": "#",
                "id": "subitem1",
                "label": "Sub-item 1",
              },
              Object {
                "href": "#",
                "id": "subitem2",
                "label": "Sub-item 2",
              },
            ],
          },
        },
        Object {
          "href": "#",
          "id": "item4",
          "label": "Item 4",
        },
      ]
    }
    classes={
      Object {
        "@media (max-width: 991px)": "@media (max-width: 991px)",
        "dropdownElement": "dropdownElement",
        "dropdownLink": "dropdownLink",
        "dropdownMenu": "dropdownMenu",
        "footer": "footer",
        "footerList": "footerList",
        "link": "link",
        "listElement": "listElement",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ul
        className="footerList"
      >
        <li
          className="listElement"
        >
          <a
            className="link"
            href="#"
            id="item1"
            onClick={undefined}
            target={undefined}
          >
            Item 1
          </a>
        </li>
        <li
          className="listElement"
        >
          <a
            className="link"
            href="#"
            id="item2"
            onClick={undefined}
            target={undefined}
          >
            Item 2
          </a>
        </li>
        <li
          className="listElement"
        >
          <a
            className="link"
            href={undefined}
            id="item3"
            onClick={[Function]}
            target={undefined}
          >
            Item 3
          </a>
          <ul
            className="dropdownMenu"
            id="item3sub"
            style={undefined}
          >
            <li
              className="dropdownElement"
            >
              <a
                className="dropdownLink"
                href="#"
                id="subitem1"
                target={undefined}
              >
                Sub-item 1
              </a>
            </li>
            <li
              className="dropdownElement"
            >
              <a
                className="dropdownLink"
                href="#"
                id="subitem2"
                target={undefined}
              >
                Sub-item 2
              </a>
            </li>
          </ul>
        </li>
        <li
          className="listElement"
        >
          <a
            className="link"
            href="#"
            id="item4"
            onClick={undefined}
            target={undefined}
          >
            Item 4
          </a>
        </li>
      </ul>,
      "className": "footer",
      "fluid": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="listElement"
          >
            <a
              className="link"
              href="#"
              id="item1"
              onClick={undefined}
              target={undefined}
            >
              Item 1
            </a>
          </li>,
          <li
            className="listElement"
          >
            <a
              className="link"
              href="#"
              id="item2"
              onClick={undefined}
              target={undefined}
            >
              Item 2
            </a>
          </li>,
          <li
            className="listElement"
          >
            <a
              className="link"
              href={undefined}
              id="item3"
              onClick={[Function]}
              target={undefined}
            >
              Item 3
            </a>
            <ul
              className="dropdownMenu"
              id="item3sub"
              style={undefined}
            >
              <li
                className="dropdownElement"
              >
                <a
                  className="dropdownLink"
                  href="#"
                  id="subitem1"
                  target={undefined}
                >
                  Sub-item 1
                </a>
              </li>
              <li
                className="dropdownElement"
              >
                <a
                  className="dropdownLink"
                  href="#"
                  id="subitem2"
                  target={undefined}
                >
                  Sub-item 2
                </a>
              </li>
            </ul>
          </li>,
          <li
            className="listElement"
          >
            <a
              className="link"
              href="#"
              id="item4"
              onClick={undefined}
              target={undefined}
            >
              Item 4
            </a>
          </li>,
        ],
        "className": "footerList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "item1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="link"
                href="#"
                id="item1"
                onClick={undefined}
                target={undefined}
              >
                Item 1
              </a>,
              undefined,
            ],
            "className": "listElement",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Item 1",
                "className": "link",
                "href": "#",
                "id": "item1",
                "onClick": undefined,
                "target": undefined,
              },
              "ref": null,
              "rendered": "Item 1",
              "type": "a",
            },
            undefined,
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "item2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="link"
                href="#"
                id="item2"
                onClick={undefined}
                target={undefined}
              >
                Item 2
              </a>,
              undefined,
            ],
            "className": "listElement",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Item 2",
                "className": "link",
                "href": "#",
                "id": "item2",
                "onClick": undefined,
                "target": undefined,
              },
              "ref": null,
              "rendered": "Item 2",
              "type": "a",
            },
            undefined,
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "item3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="link"
                href={undefined}
                id="item3"
                onClick={[Function]}
                target={undefined}
              >
                Item 3
              </a>,
              <ul
                className="dropdownMenu"
                id="item3sub"
                style={undefined}
              >
                <li
                  className="dropdownElement"
                >
                  <a
                    className="dropdownLink"
                    href="#"
                    id="subitem1"
                    target={undefined}
                  >
                    Sub-item 1
                  </a>
                </li>
                <li
                  className="dropdownElement"
                >
                  <a
                    className="dropdownLink"
                    href="#"
                    id="subitem2"
                    target={undefined}
                  >
                    Sub-item 2
                  </a>
                </li>
              </ul>,
            ],
            "className": "listElement",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Item 3",
                "className": "link",
                "href": undefined,
                "id": "item3",
                "onClick": [Function],
                "target": undefined,
              },
              "ref": null,
              "rendered": "Item 3",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="dropdownElement"
                  >
                    <a
                      className="dropdownLink"
                      href="#"
                      id="subitem1"
                      target={undefined}
                    >
                      Sub-item 1
                    </a>
                  </li>,
                  <li
                    className="dropdownElement"
                  >
                    <a
                      className="dropdownLink"
                      href="#"
                      id="subitem2"
                      target={undefined}
                    >
                      Sub-item 2
                    </a>
                  </li>,
                ],
                "className": "dropdownMenu",
                "id": "item3sub",
                "style": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "subitem1",
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      className="dropdownLink"
                      href="#"
                      id="subitem1"
                      target={undefined}
                    >
                      Sub-item 1
                    </a>,
                    "className": "dropdownElement",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sub-item 1",
                      "className": "dropdownLink",
                      "href": "#",
                      "id": "subitem1",
                      "target": undefined,
                    },
                    "ref": null,
                    "rendered": "Sub-item 1",
                    "type": "a",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "subitem2",
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      className="dropdownLink"
                      href="#"
                      id="subitem2"
                      target={undefined}
                    >
                      Sub-item 2
                    </a>,
                    "className": "dropdownElement",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sub-item 2",
                      "className": "dropdownLink",
                      "href": "#",
                      "id": "subitem2",
                      "target": undefined,
                    },
                    "ref": null,
                    "rendered": "Sub-item 2",
                    "type": "a",
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "item4",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="link"
                href="#"
                id="item4"
                onClick={undefined}
                target={undefined}
              >
                Item 4
              </a>,
              undefined,
            ],
            "className": "listElement",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Item 4",
                "className": "link",
                "href": "#",
                "id": "item4",
                "onClick": undefined,
                "target": undefined,
              },
              "ref": null,
              "rendered": "Item 4",
              "type": "a",
            },
            undefined,
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ul
          className="footerList"
        >
          <li
            className="listElement"
          >
            <a
              className="link"
              href="#"
              id="item1"
              onClick={undefined}
              target={undefined}
            >
              Item 1
            </a>
          </li>
          <li
            className="listElement"
          >
            <a
              className="link"
              href="#"
              id="item2"
              onClick={undefined}
              target={undefined}
            >
              Item 2
            </a>
          </li>
          <li
            className="listElement"
          >
            <a
              className="link"
              href={undefined}
              id="item3"
              onClick={[Function]}
              target={undefined}
            >
              Item 3
            </a>
            <ul
              className="dropdownMenu"
              id="item3sub"
              style={undefined}
            >
              <li
                className="dropdownElement"
              >
                <a
                  className="dropdownLink"
                  href="#"
                  id="subitem1"
                  target={undefined}
                >
                  Sub-item 1
                </a>
              </li>
              <li
                className="dropdownElement"
              >
                <a
                  className="dropdownLink"
                  href="#"
                  id="subitem2"
                  target={undefined}
                >
                  Sub-item 2
                </a>
              </li>
            </ul>
          </li>
          <li
            className="listElement"
          >
            <a
              className="link"
              href="#"
              id="item4"
              onClick={undefined}
              target={undefined}
            >
              Item 4
            </a>
          </li>
        </ul>,
        "className": "footer",
        "fluid": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="listElement"
            >
              <a
                className="link"
                href="#"
                id="item1"
                onClick={undefined}
                target={undefined}
              >
                Item 1
              </a>
            </li>,
            <li
              className="listElement"
            >
              <a
                className="link"
                href="#"
                id="item2"
                onClick={undefined}
                target={undefined}
              >
                Item 2
              </a>
            </li>,
            <li
              className="listElement"
            >
              <a
                className="link"
                href={undefined}
                id="item3"
                onClick={[Function]}
                target={undefined}
              >
                Item 3
              </a>
              <ul
                className="dropdownMenu"
                id="item3sub"
                style={undefined}
              >
                <li
                  className="dropdownElement"
                >
                  <a
                    className="dropdownLink"
                    href="#"
                    id="subitem1"
                    target={undefined}
                  >
                    Sub-item 1
                  </a>
                </li>
                <li
                  className="dropdownElement"
                >
                  <a
                    className="dropdownLink"
                    href="#"
                    id="subitem2"
                    target={undefined}
                  >
                    Sub-item 2
                  </a>
                </li>
              </ul>
            </li>,
            <li
              className="listElement"
            >
              <a
                className="link"
                href="#"
                id="item4"
                onClick={undefined}
                target={undefined}
              >
                Item 4
              </a>
            </li>,
          ],
          "className": "footerList",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "item1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="link"
                  href="#"
                  id="item1"
                  onClick={undefined}
                  target={undefined}
                >
                  Item 1
                </a>,
                undefined,
              ],
              "className": "listElement",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Item 1",
                  "className": "link",
                  "href": "#",
                  "id": "item1",
                  "onClick": undefined,
                  "target": undefined,
                },
                "ref": null,
                "rendered": "Item 1",
                "type": "a",
              },
              undefined,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "item2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="link"
                  href="#"
                  id="item2"
                  onClick={undefined}
                  target={undefined}
                >
                  Item 2
                </a>,
                undefined,
              ],
              "className": "listElement",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Item 2",
                  "className": "link",
                  "href": "#",
                  "id": "item2",
                  "onClick": undefined,
                  "target": undefined,
                },
                "ref": null,
                "rendered": "Item 2",
                "type": "a",
              },
              undefined,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "item3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="link"
                  href={undefined}
                  id="item3"
                  onClick={[Function]}
                  target={undefined}
                >
                  Item 3
                </a>,
                <ul
                  className="dropdownMenu"
                  id="item3sub"
                  style={undefined}
                >
                  <li
                    className="dropdownElement"
                  >
                    <a
                      className="dropdownLink"
                      href="#"
                      id="subitem1"
                      target={undefined}
                    >
                      Sub-item 1
                    </a>
                  </li>
                  <li
                    className="dropdownElement"
                  >
                    <a
                      className="dropdownLink"
                      href="#"
                      id="subitem2"
                      target={undefined}
                    >
                      Sub-item 2
                    </a>
                  </li>
                </ul>,
              ],
              "className": "listElement",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Item 3",
                  "className": "link",
                  "href": undefined,
                  "id": "item3",
                  "onClick": [Function],
                  "target": undefined,
                },
                "ref": null,
                "rendered": "Item 3",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li
                      className="dropdownElement"
                    >
                      <a
                        className="dropdownLink"
                        href="#"
                        id="subitem1"
                        target={undefined}
                      >
                        Sub-item 1
                      </a>
                    </li>,
                    <li
                      className="dropdownElement"
                    >
                      <a
                        className="dropdownLink"
                        href="#"
                        id="subitem2"
                        target={undefined}
                      >
                        Sub-item 2
                      </a>
                    </li>,
                  ],
                  "className": "dropdownMenu",
                  "id": "item3sub",
                  "style": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "subitem1",
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        className="dropdownLink"
                        href="#"
                        id="subitem1"
                        target={undefined}
                      >
                        Sub-item 1
                      </a>,
                      "className": "dropdownElement",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Sub-item 1",
                        "className": "dropdownLink",
                        "href": "#",
                        "id": "subitem1",
                        "target": undefined,
                      },
                      "ref": null,
                      "rendered": "Sub-item 1",
                      "type": "a",
                    },
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": "subitem2",
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        className="dropdownLink"
                        href="#"
                        id="subitem2"
                        target={undefined}
                      >
                        Sub-item 2
                      </a>,
                      "className": "dropdownElement",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Sub-item 2",
                        "className": "dropdownLink",
                        "href": "#",
                        "id": "subitem2",
                        "target": undefined,
                      },
                      "ref": null,
                      "rendered": "Sub-item 2",
                      "type": "a",
                    },
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "item4",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="link"
                  href="#"
                  id="item4"
                  onClick={undefined}
                  target={undefined}
                >
                  Item 4
                </a>,
                undefined,
              ],
              "className": "listElement",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Item 4",
                  "className": "link",
                  "href": "#",
                  "id": "item4",
                  "onClick": undefined,
                  "target": undefined,
                },
                "ref": null,
                "rendered": "Item 4",
                "type": "a",
              },
              undefined,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Footer styles matches the snapshot 1`] = `
Object {
  "@media (max-width: 991px)": Object {
    "footer": Object {
      "display": "none !important",
    },
  },
  "dropdownElement": Object {
    "float": "left",
    "listStyle": "none",
  },
  "dropdownLink": Object {
    "&:hover": Object {
      "color": "#ffffff",
      "textDecoration": "none",
    },
    "color": "#cdcdcd",
    "display": "block",
    "lineHeight": "1.42857143",
    "padding": "3px 18px",
    "textDecoration": "none",
  },
  "dropdownMenu": Object {
    "background": "#242424",
    "borderRadius": "4px",
    "bottom": "calc(100% + 10px)",
    "boxShadow": "0 6px 12px rgba(0,0,0,.175)",
    "display": "none",
    "fontSize": "12px",
    "left": "50%",
    "listStyle": "none",
    "minWidth": "160px",
    "padding": "5px 0",
    "position": "absolute",
    "textAlign": "left",
    "transform": "translateX(-50%)",
    "zIndex": "2",
  },
  "footer": Object {
    "backgroundColor": "#242424",
    "color": "#cdcdcd",
    "fontFamily": "'Nunito Sans', sans-serif",
    "fontSize": "12px",
    "height": "25px",
    "margin": "0",
    "padding": "0 40px",
  },
  "footerList": Object {
    "listStyle": "none",
    "margin": "0",
    "padding": "4px 0 0 0",
    "width": "auto",
  },
  "link": Object {
    "&:hover": Object {
      "color": "#ffffff",
      "textDecoration": "none",
    },
    "color": "#cdcdcd",
    "cursor": "pointer",
    "textDecoration": "none",
  },
  "listElement": Object {
    "float": "left",
    "listStyle": "none",
    "margin": "0 50px 0 0",
    "padding": "0",
    "position": "relative",
  },
}
`;

exports[`FooterJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Footer)
    buttons={
      Array [
        Object {
          "href": "#",
          "id": "item1",
          "label": "Item 1",
        },
        Object {
          "href": "#",
          "id": "item2",
          "label": "Item 2",
        },
        Object {
          "id": "item3",
          "label": "Item 3",
          "sub": Object {
            "id": "item3sub",
            "items": Array [
              Object {
                "href": "#",
                "id": "subitem1",
                "label": "Sub-item 1",
              },
              Object {
                "href": "#",
                "id": "subitem2",
                "label": "Sub-item 2",
              },
            ],
          },
        },
        Object {
          "href": "#",
          "id": "item4",
          "label": "Item 4",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "buttons": Array [
        Object {
          "href": "#",
          "id": "item1",
          "label": "Item 1",
        },
        Object {
          "href": "#",
          "id": "item2",
          "label": "Item 2",
        },
        Object {
          "id": "item3",
          "label": "Item 3",
          "sub": Object {
            "id": "item3sub",
            "items": Array [
              Object {
                "href": "#",
                "id": "subitem1",
                "label": "Sub-item 1",
              },
              Object {
                "href": "#",
                "id": "subitem2",
                "label": "Sub-item 2",
              },
            ],
          },
        },
        Object {
          "href": "#",
          "id": "item4",
          "label": "Item 4",
        },
      ],
      "classes": Object {
        "dropdownElement": "Footer-dropdownElement-0-1-6",
        "dropdownLink": "Footer-dropdownLink-0-1-7",
        "dropdownMenu": "Footer-dropdownMenu-0-1-5",
        "footer": "Footer-footer-0-1-1",
        "footerList": "Footer-footerList-0-1-2",
        "link": "Footer-link-0-1-4",
        "listElement": "Footer-listElement-0-1-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "buttons": Array [
          Object {
            "href": "#",
            "id": "item1",
            "label": "Item 1",
          },
          Object {
            "href": "#",
            "id": "item2",
            "label": "Item 2",
          },
          Object {
            "id": "item3",
            "label": "Item 3",
            "sub": Object {
              "id": "item3sub",
              "items": Array [
                Object {
                  "href": "#",
                  "id": "subitem1",
                  "label": "Sub-item 1",
                },
                Object {
                  "href": "#",
                  "id": "subitem2",
                  "label": "Sub-item 2",
                },
              ],
            },
          },
          Object {
            "href": "#",
            "id": "item4",
            "label": "Item 4",
          },
        ],
        "classes": Object {
          "dropdownElement": "Footer-dropdownElement-0-1-6",
          "dropdownLink": "Footer-dropdownLink-0-1-7",
          "dropdownMenu": "Footer-dropdownMenu-0-1-5",
          "footer": "Footer-footer-0-1-1",
          "footerList": "Footer-footerList-0-1-2",
          "link": "Footer-link-0-1-4",
          "listElement": "Footer-listElement-0-1-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
